core/src/main/java/io/grpc/internal/DnsNameResolver.java

```diff
@@ -293,19 +293,24 @@ public void run() {
             Status.UNAVAILABLE.withDescription("Unable to resolve host " + host).withCause(e));
         return;
       }
+      if (resolutionResults.addresses.isEmpty() && resolutionResults.balancerAddresses.isEmpty()) {
+        savedListener.onError(Status.UNAVAILABLE.withDescription(
+            "No DNS backend or balancer addresses found for " + host));
+        return;
+      }
       // Each address forms an EAG
       List<EquivalentAddressGroup> servers = new ArrayList<>();
       for (InetAddress inetAddr : resolutionResults.addresses) {
         servers.add(new EquivalentAddressGroup(new InetSocketAddress(inetAddr, port)));
       }
-      servers.addAll(resolutionResults.balancerAddresses);
-      if (servers.isEmpty()) {
-        savedListener.onError(Status.UNAVAILABLE.withDescription(
-            "No DNS backend or balancer addresses found for " + host));
-        return;
-      }
 
-      ResolutionResult.Builder resultBuilder = ResolutionResult.newBuilder().setAddresses(servers);
+      ResolutionResult.Builder resultBuilder =
+          ResolutionResult.newBuilder()
+              .setAddresses(servers)
+              .setAttributes(
+                  Attributes.newBuilder()
```

Comment:
this will be overwritten in some case where it execute L330. it looks like we should have both.

Reply:
Fixed.